e2e/test/iothub/properties/PropertiesE2ETests.cs

```diff
@@ -0,0 +1,483 @@
+            {
+                Assert.AreEqual<string>(propFromCollection, propValue);
+            } else
+            {
+                Assert.Fail("Property not found in ClientProperties");
+            }
+            await deviceClient.CloseAsync().ConfigureAwait(false);
+            await registryManager.CloseAsync().ConfigureAwait(false);
+        }
+
+        private async Task Properties_DeviceSetsPropertyAndServiceReceivesItAsync(Client.TransportType transport)
+        {
+            var propName = Guid.NewGuid().ToString();
+            var propValue = Guid.NewGuid().ToString();
+
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            var patch = new ClientPropertyCollection();
+            patch[propName] = propValue;
+            await deviceClient.UpdateClientPropertiesAsync(patch).ConfigureAwait(false);
+            await deviceClient.CloseAsync().ConfigureAwait(false);
+
+            Twin serviceTwin = await registryManager.GetTwinAsync(testDevice.Id).ConfigureAwait(false);
+            Assert.AreEqual<string>(serviceTwin.Properties.Reported[propName].ToString(), propValue);
+
+            Logger.Trace("verified " + serviceTwin.Properties.Reported[propName].ToString() + "=" + propValue);
+        }
+
+        private async Task Properties_ServiceDoesNotCreateNullPropertyInCollectionAsync(Client.TransportType transport)
+        {
+            var propName1 = Guid.NewGuid().ToString();
+            var propName2 = Guid.NewGuid().ToString();
+            var propEmptyValue = "{}";
+
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            await deviceClient
+                .UpdateClientPropertiesAsync(
+                    new ClientPropertyCollection
+                    {
+                        [propName1] = null
+                    })
+                .ConfigureAwait(false);
+            Twin serviceTwin = await registryManager.GetTwinAsync(testDevice.Id).ConfigureAwait(false);
+            Assert.IsFalse(serviceTwin.Properties.Reported.Contains(propName1));
+
+            await deviceClient
+                .UpdateClientPropertiesAsync(
+                    new ClientPropertyCollection
+                    {
+                        [propName1] = new Dictionary<string, object>
+                        {
+                            [propName2] = null
+                        }
```

Comment:
interesting, I wasn't aware that propName2 wouldn't make it to the final twin

Reply:
I thought the removal of "null" value properties referred to only root level properties, and not nested ones.